lib/jekyll-remote-theme/downloader.rb

```diff
@@ -37,31 +48,55 @@ def zip_file
         @zip_file ||= Tempfile.new([TEMP_PREFIX, ".zip"], :binmode => true)
       end
 
-      def download
-        Jekyll.logger.debug LOG_KEY, "Downloading #{zip_url} to #{zip_file.path}"
-        Net::HTTP.start(zip_url.host, zip_url.port, :use_ssl => true) do |http|
-          http.request(request) do |response|
-            raise_unless_sucess(response)
-            enforce_max_file_size(response.content_length)
-            response.read_body do |chunk|
-              zip_file.write chunk
+      def create_request(url)
+        req = Net::HTTP::Get.new url.path
+
+        remote_headers&.each do |key, value|
+          req[key] = value unless value.nil?
+        end
+
+        req
+      end
+
+      def handle_response(response)
+        raise_unless_success(response)
+        enforce_max_file_size(response.content_length)
+        response.read_body do |chunk|
+          zip_file.write chunk
+        end
+      end
+
+      def fetch(uri_str, limit = 10)
+        Jekyll.logger.debug LOG_KEY, "Finding redirect of #{uri_str}"
+
+        raise DownloadError, "Too many redirect" if limit.zero?
+
+        url = URI.parse(uri_str)
+        req = create_request(url)
+
+        Net::HTTP.start(url.host, url.port, :use_ssl => true) do |http|
+          http.request(req) do |response|
+            case response
+            when Net::HTTPSuccess
+              handle_response(response)
+            when Net::HTTPRedirection
+              fetch(response["location"], limit - 1)
```

Comment:
Our organization is using a custom DNS, and the REST Api are creating some redirections.  This new download code supports HTTP Redirections to a maximum of 10 hops.